modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/utils/JavaParserUtils.java

```diff
@@ -0,0 +1,385 @@
+        if (parameterType.isAssignableTo(ConnectionProvider.class)
+            || parameterType.isAssignableTo(org.mule.sdk.api.connectivity.ConnectionProvider.class)) {
+          calculatedMMV = maxMMV(calculatedMMV, calculateConnectionProviderMinMuleVersion(parameterType));
+        }
+      }
+      if (annotation.isSameType(Config.class) || annotation.isSameType(org.mule.sdk.api.annotation.param.Config.class)) {
+        calculatedMMV =
+            maxMMV(calculatedMMV, calculateConfigMinMuleVersion(parameterType, getEnforcedMinMuleVersion(annotation)));
+      }
+      calculatedMMV = maxMMV(calculatedMMV, getEnforcedMinMuleVersion(annotation));
+    }
+    // Parse sdk fields (this also catches automatically injected fields)
+    if (belongsToSdkPackages(parameterType.getTypeName())) {
+      calculatedMMV = maxMMV(calculatedMMV, getEnforcedMinMuleVersion(parameterType));
+    }
+    return calculatedMMV;
+  }
+
+  private static MuleVersion calculateOperationParameterMinMuleVersion(ExtensionParameter operationParameter) {
+    MuleVersion calculatedMMV = FIRST_MULE_VERSION;
+    for (Type annotation : operationParameter.getAnnotations()) {
+      if (annotation.isSameType(ParameterGroup.class)
+          || annotation.isSameType(org.mule.runtime.extension.api.annotation.param.ParameterGroup.class)) {
+        calculatedMMV = maxMMV(calculatedMMV, calculateParameterContainerMinMuleVersion(operationParameter.getType()));
+      }
+      if (annotation.isSameType(Config.class) || annotation.isSameType(org.mule.sdk.api.annotation.param.Config.class)) {
+        calculatedMMV =
+            maxMMV(calculatedMMV,
+                   calculateConfigMinMuleVersion(operationParameter.getType(), getEnforcedMinMuleVersion(annotation)));
+      }
+      calculatedMMV = maxMMV(calculatedMMV, getEnforcedMinMuleVersion(annotation));
+    }
+    // Parse sdk fields (this also catches automatically injected fields)
+    if (belongsToSdkPackages(operationParameter.getType().getTypeName())) {
+      calculatedMMV = maxMMV(calculatedMMV, getEnforcedMinMuleVersion(operationParameter.getType()));
+    }
+    return calculatedMMV;
+  }
+
+  private static MuleVersion calculateParameterContainerMinMuleVersion(Type containerType) {
+    Optional<MuleVersion> minMuleVersionAnnotation = getMinMuleVersionFromAnnotations(containerType);
+    if (minMuleVersionAnnotation.isPresent()) {
+      return minMuleVersionAnnotation.get();
+    }
+    MuleVersion calculatedMMV = FIRST_MULE_VERSION;
+    for (Type annotation : containerType.getAnnotations()) {
+      calculatedMMV = maxMMV(calculatedMMV, getEnforcedMinMuleVersion(annotation));
+    }
+    for (FieldElement field : containerType.getFields()) {
+      calculatedMMV = maxMMV(calculatedMMV, calculateFieldMinMuleVersion(field));
+    }
+    return calculatedMMV;
+  }
+
+  private static List<MethodElement> getAnnotatedSourceMethods(Type source) {
```

Comment:
why is this method necessary? Why not just analysing all the methods in the source, which would also caught methods from other interfaces. Another con of this method is that it needs changes each time we add a new feature or annotation

Reply:
Was trying to be more surgical, but agreed that better parse all methods.